src/main/java/net/syscon/elite/repository/impl/InmateRepositoryImpl.java

```diff
@@ -888,6 +890,24 @@ public void updateActiveCategoryNextReviewDate(final long bookingId, final Local
                 OFFENDER_BASIC_DETAILS_MAPPER);
     }
 
+    @Override
+    public Optional<ImprisonmentStatus> getImprisonmentStatus(final long bookingId) {
+        ImprisonmentStatus imprisonmentStatus;
+        try {
+            imprisonmentStatus = jdbcTemplate.queryForObject(
+                    getQuery("GET_IMPRISONMENT_STATUS"),
+                    createParams("bookingId", bookingId),
+                    IMPRISONMENT_STATUS_MAPPER);
+            if (imprisonmentStatus != null) {
```

Comment:
this seems a bit weird - when would it return null and when would it throw `EmptyResultDataAccessException`? 
are these both trying to handle not getting back data?